### PR TITLE
Scan and HTTP request events by planner

### PR DESCRIPTION
**Pull request by brechtvdv**

### CODE REVIEW

File: lib/http/HttpConnectionsStream.js [hunk 1]
```diff
@@ -35,6 +35,7 @@ HttpConnectionsStream.prototype.close = function () {
 
 HttpConnectionsStream.prototype._fetchNextPage = function () {
   var self = this;
+  this.emit('request');
```
Comment (pietercolpaert) [inline lib/http/HttpConnectionsStream.js:38]: So, you're not emitting an event when fetching the first page? I'd put the emitter in the HttpFetcher class when on .get

File: lib/lc-client.js [hunk 2]
```diff
@@ -29,6 +28,13 @@ Client.prototype.query = function (q, cb) {
     planner.on("result", function () {
       self._fetcher.close();
     });
+    var amountConnections = 0;
+    connectionsStream.on('data', function() {
+      planner.emit('scan'); // connection gets scanned
+    });
+    connectionsStream.on('request', function() {
```
Comment (pietercolpaert) [inline lib/lc-client.js:35]: would check on `this._http.on('request'...)` instead

File: lib/http/HttpConnectionsStream.js (hunk 1 above)
Comment (brechtvdv) [inline lib/http/HttpConnectionsStream.js:38]: Forgot the first page. Tried to put it in HttpFetcher by extending EventEmitter, but got `undefined is not a function` when registering an eventlistener in Fetcher.js: 

``` javascript
this.http = new HttpFetcher(20);
this.http.on('request', function() { ... });
```

this.http does exist when debugging.. Any idea?

Comment (pietercolpaert) [inline lib/http/HttpConnectionsStream.js:38]: Yes: you need to make Fetcher.js inherit from EventEmitter. Only then you'll be able to emit events :)

**Comment by pietercolpaert** [comment]
I've fixed this myself in the 0.0.3 release :)

**Comment by brechtvdv** [comment]
Great!

**Pull request closed without merging**